internal/config/config.go

```diff
@@ -386,18 +387,23 @@ func fromFile(fileHandler file.Handler, name string) (*Config, error) {
 // 3. Read secrets from environment variables.
 // 4. Validate config. If `--force` is set the version validation will be disabled and any version combination is allowed.
 func New(fileHandler file.Handler, name string, force bool) (*Config, error) {
-	return NewWithClient(fileHandler, name, fetcher.NewHTTPClient(), force)
+	return NewWithFetcher(fileHandler, name, fetcher.NewConfigAPIFetcher(), force)
 }
 
-// NewWithClient is New with a custom HTTP client.
-func NewWithClient(fileHandler file.Handler, name string, client fetcher.HTTPClient, force bool) (*Config, error) {
+// NewWithFetcher is New with a custom fetcher.
+func NewWithFetcher(fileHandler file.Handler, name string, fetcher fetcher.ConfigAPIFetcher, force bool) (*Config, error) {
```

Comment:
Since using `config.New()` over `config.NewWithFetcher()` is incorrect in most cases, any objections against removing the `config.NewWithFetcher()` and just requiring `config.New()` to take a ConfigAPIFetcher?

My reasoning:
* `config.New()` can only be called if we are sure that the function making the call will never be accessed in a unit test.
* In a perfect world, all code is tested somehow, so those instances should use `config.NewWithFetcher()` instead.
* A developer is likely to chose `config.New()` over `config.NewWithFetcher()` as this looks to be the default initializer for the config. However, this is most likely the wrong function for their use case, as has explained above

Reply:
I agree, I was thinking the same but lost it again amidst all the refactoring 🙈 Applying the changes now

Reply:
I agree. I was thinking the same but lost it again amidst all the refactoring 🙈 Applying the changes now